cs-minibot-platform-src/src/main/java/examples/gobot/GoBot.java

```diff
@@ -0,0 +1,375 @@
+                            } else if (course.getMiddleArea().contains(vc.x, vc.y)) {
+                                if (!reachedMiddle) {
+                                    this.reachedMiddle = true;
+                                }
+                            }
+                        } else {
+                            System.out.println("Out of bounds");
+                            inTrack = false;
+                        }
+                    }
+                }
+            }
+            else if (botState == BOT_PLAYING) { //ASSUMING CCL TRACK
+                if (lapsDone >= numLaps) {
+                    // Finished!
+                    botState = WAITING;
+                    botTime = totalTime();
+                    lapTimes.clear();
+                    lapsDone = 0;
+                    printState();
+                }
+                else if (aiType == 0){
+                    List<VisionObject> vl =  BaseStation.getInstance().getVisionManager()
+                            .getAllLocationData();
+                    if (vl.size() != 0) {
+                        VisionCoordinate vc = vl.get(0).coord;
+                        inTrack = this.course.isInsideTrack(vc);
+                        if (navigator.destinationReached()) {
+                            //int max = BaseHTTPInterface.innerTrackCoords
+                            // .size(); for tracing inner track
+                            int max = BaseHTTPInterface.advancedAI.size();
+                            if (max != 0) {
+                                //navigator.goToDestination(BaseHTTPInterface
+                                // .innerTrackCoords.get(index)); for tracing
+                                // inner track
+                                navigator.goToDestination(BaseHTTPInterface
+                                        .advancedAI.get(index));
+                                System.out.println(navigator.destination);
+                                index = (index + 1) % max;
+                            }
+                        }
+                        if (course.getStartArea().contains(vc.x, vc.y)) {
+                            if (!crossedLapLine && !reachedMiddle) {
+                                this.crossedLapLine = true;
+                            } else if (crossedLapLine && reachedMiddle) {
+                                this.lapsDone++;
+                                this.reachedMiddle = false;
+                                this.lapTimes.add(System.nanoTime());
+                                System.out.println(lapTimes());
+                            }
+                        } else if (course.getMiddleArea().contains(vc.x, vc.y)) {
+                            if (!reachedMiddle) {
+                                this.reachedMiddle = true;
+                            }
+                        }
+                    }
+                }
+                else{
+                    navigator.run();
```

Comment:
Yes, I believe this works. I had it in my code on demo day. I believe it just runs the algorithm. If it is advanced AI, it will control it based on that.